src/opm/parser/eclipse/EclipseState/Schedule/Well/WellConnections.cpp

```diff
@@ -195,10 +195,17 @@ inline std::array< size_t, 3> directionIndices(const Opm::Connection::Direction
         return update;
     }
 
-    void WellConnections::applyWellPIScaling(const double scaleFactor)
+    void WellConnections::applyWellPIScaling(const double       scaleFactor,
+                                             std::vector<bool>& scalingApplicable)
     {
-        for (auto& conn : this->m_connections)
-            conn.applyWellPIScaling(scaleFactor);
+        scalingApplicable.resize(std::max(scalingApplicable.size(), this->m_connections.size()), true);
```

Comment:
The `scalingApplicable`vector seems unused to me - always `true` for all elements?

Reply:
It is an in/out parameter.  The overall process is somewhat subtle however and I'd very much like a better suggestion.

The `resize()` call (not `assign()`) is to handle new connections being added during the time series, and all connections are **initially** eligible for `WELPI` scaling.  This also means the caller, currently the new member function `Schedule::applyWellProdIndexScaling()` only, can start by passing an empty `scalingApplicable` array.  We then possibly turn off the eligibility status in the statement
```C++
scalingApplicable[i] = conn.applyWellScaling(scaleFactor);
```
at which point it remains turned off for the remainder of the time series assuming the caller passes the same `scalingApplicable` array into the next call to `WellConnections::applyWellPIScaling()`.

In other words, the `scalingApplicable` provides a way to communicate eligibility status between the unique `WellConnections` instances.

Reply:
Ok - I'll have another look, but to me it looked like there was never an assignment to `false`?

Reply:
> it looked like there was never an assignment to `false`?

I suppose this too is a bit subtle.  [`Connection::applyWellPIScaling()`](https://github.com/OPM/opm-common/blob/91bb168ee58d40875c0f2598d4af205d50166767/src/opm/parser/eclipse/EclipseState/Schedule/Well/Connection.cpp#L259-L265) returns `false` when the connection is not eligible for WELPI scaling.  In other words, the [update loop](https://github.com/OPM/opm-common/blob/91bb168ee58d40875c0f2598d4af205d50166767/src/opm/parser/eclipse/EclipseState/Schedule/Well/WellConnections.cpp#L204-L208) also assigns the `false` values if needed:
```C++
for (auto& conn : this->m_connections) {
    if (scalingApplicable[i])
        scalingApplicable[i] = conn.applyWellPIScaling(scaleFactor);
    ++i;
}
```